### PR TITLE
(core) introduce typescript, modularize app data sources

### PR DESCRIPTION
**Pull request by anotherchrisberry**
Bringing in Typescript, refactoring the Application into a proper class with actual documentation.

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/// <reference path="globals/angular-mocks/index.d.ts" />
```
Comment (icfantv) [inline typings/index.d.ts]: don't commit this file.  it's generated by `typings`.

File: package.json [hunk 2]
```diff
@@ -51,6 +51,7 @@
     "build": "node ./node_modules/webpack/bin/webpack.js",
     "start": "node ./node_modules/webpack-dev-server/bin/webpack-dev-server.js",
     "test": "./node_modules/karma/bin/karma start",
+    "preinstall": "npm install -g typings && typings install",
```
Comment (icfantv) [inline package.json]: i could be wrong, but i _think_ this can be changed to just `"postinstall": typings install`.

Comment (anotherchrisberry) [inline package.json]: Yep, good call, that works.

File: typings/index.d.ts (hunk 1 above)
Comment (anotherchrisberry) [inline typings/index.d.ts]: You're right - thanks for catching.

**Pull request merged**